### PR TITLE
ASM-3534 fix sporadic idrac firmware reboot job failures

### PR DESCRIPTION
**Pull request by gavin-scott**
The Tempfile being passed to the idrac reboot job was being
garbage collected away in some cases. Change to use the common
ASM::WsMan.reboot method.

### CODE REVIEW

**Comment by gavin-scott** [comment]
Deployed changes on @sushilrai 's appliance, he is running a test.

**Comment by logankimmel** [comment]
@gavin-scott ****Don't Merge this.  Will break all firmware updates

File: lib/puppet/provider/idrac_fw_installfromuri/wsman.rb [hunk 1]
```diff
@@ -113,8 +114,8 @@ def update(firmware_list)
     reboot_id = nil
     if reboot_required
       if @force_restart
-        reboot_config_file_path = create_reboot_config_file
-        reboot_id = create_reboot_job(reboot_config_file_path)
+        require 'asm/wsman'
```
Comment (logankimmel) [inline lib/puppet/provider/idrac_fw_installfromuri/wsman.rb]: This needs to return the reboot_id for us to check it below or will fail

File: lib/puppet/provider/idrac_fw_installfromuri/wsman.rb [hunk 2]
```diff
@@ -113,8 +114,8 @@ def update(firmware_list)
...
+        ASM::WsMan.reboot(transport, Puppet)
```
Comment (logankimmel) [inline lib/puppet/provider/idrac_fw_installfromuri/wsman.rb]: reboot_id = ASM::WsMan.reboot(transport, Puppet)

**Comment by gavin-scott** [comment]
Updated with @logankimmel 's fix, but it will require a change to the ASM::WsMan.reboot method to return the job id:

https://github.com/dell-asm/asm-deployer/blob/cb7ce03890c7bde157d2018e2b19fb4067a82e14/lib/asm/wsman.rb#L118

DO NOT MERGE! Requires the above to be changed in asm-deployer

**Comment by logankimmel** [comment]
@gavin-scott Let's talk about this.  I don't think the proposed solution is going to work for several reasons.

**Comment by gavin-scott** [comment]
@logankimmel pointed out to me that the ASM::WsMan.reboot method won't work because it initiates the job. This module needs to bundle it up with the other firmware update jobs before initiating.

**Pull request closed without merging**